spipe/main.c

```diff
@@ -309,26 +324,29 @@ main(int argc, char * argv[])
 	/* Success! */
 	exit(0);
 
-err6:
+err7:
 	if ((rc = pthread_cancel(ET.threads[0])) != 0)
 		warn0("pthread_cancel: %s", strerror(rc));
 	if ((rc = pthread_join(ET.threads[0], NULL)) != 0)
 		warn0("pthread_join: %s", strerror(rc));
-err5:
+err6:
 	if ((rc = pthread_cancel(ET.threads[1])) != 0)
 		warn0("pthread_cancel: %s", strerror(rc));
 	if ((rc = pthread_join(ET.threads[1], NULL)) != 0)
 		warn0("pthread_join: %s", strerror(rc));
-err4:
+err5:
 	proto_conn_drop(conn_cookie, PROTO_CONN_CANCELLED);
-err3:
+err4:
 	if (s[1] != -1)
 		close(s[1]);
 	close(s[0]);
-err2:
+err3:
 	proto_crypt_secret_free(K);
-err1:
+err2:
 	sock_addr_freelist(sas_t);
+err1:
+	if (opt_b)
+		sock_addr_free(sa_b);
```

Comment:
If we set `sas_b = NULL` by default, this can just be `sock_addr_freelist(sas_b)` (since `free(NULL)` is safe).

Reply:
Done for both spipe and spiped.